batch/batch/worker/worker.py

```diff
@@ -425,138 +490,271 @@ async def batch_worker_access_token(self):
+                'destination': '/sys',
+                'type': 'sysfs',
+                'options': ['nosuid', 'noexec', 'nodev', 'ro'],
+            },
+            {
+                'source': 'cgroup',
+                'destination': '/sys/fs/cgroup',
+                'type': 'cgroup',
+                'options': ['nosuid', 'noexec', 'nodev', 'ro'],
+            },
+            {
+                'source': 'mqueue',
+                'destination': '/dev/mqueue',
+                'type': 'mqueue',
+                'options': ['nosuid', 'noexec', 'nodev'],
+            },
+            {
+                'source': 'shm',
+                'destination': '/dev/shm',
+                'type': 'tmpfs',
+                'options': ['nosuid', 'noexec', 'nodev', 'mode=1777', 'size=67108864'],
+            },
+            {
+                'source': f'/etc/netns/{self.netns.network_name}/resolv.conf',
+                'destination': '/etc/resolv.conf',
+                'type': 'none',
+                'options': ['rbind', 'ro'],
+            },
+            {
+                'source': f'/etc/netns/{self.netns.network_name}/hosts',
+                'destination': '/etc/hosts',
+                'type': 'none',
+                'options': ['rbind', 'ro'],
+            },
+        ]
 
-    async def get_log(self):
-        if self.container:
-            return await self.get_container_log()
-        return self.log
+    def _env(self):
+        env = self.image_config['Config']['Env'] + self.spec.get('env', [])
+        if self.port is not None:
+            assert self.host_port is not None
+            env.append(f'HAIL_BATCH_WORKER_PORT={self.host_port}')
+            env.append(f'HAIL_BATCH_WORKER_IP={IP_ADDRESS}')
+        return env
 
     async def delete_container(self):
-        if self.container:
+        if self.container_is_running():
             try:
-                log.info(f'{self}: deleting container')
-                await docker_call_retry(MAX_DOCKER_OTHER_OPERATION_SECS, f'{self}')(stop_container, self.container)
-                # v=True deletes anonymous volumes created by the container
-                await docker_call_retry(MAX_DOCKER_OTHER_OPERATION_SECS, f'{self}')(
-                    delete_container, self.container, v=True
-                )
-                self.container = None
+                log.info(f'{self} container is still running, killing crun process')
+                self.proc.kill()
             except asyncio.CancelledError:
                 raise
             except Exception:
                 log.warning('while deleting container, ignoring', exc_info=True)
+        self.proc = None
 
         if self.host_port is not None:
             port_allocator.free(self.host_port)
             self.host_port = None
 
+        if self.netns:
+            network_allocator.free(self.netns)
+            self.netns = None
+
+        # FIXME I am not sure why this does not always exist
```

Comment:
I'm disinclined to leave FIXMEs in the code. Can we get to the bottom of this?

Reply:
yes, that's the last todo once buildkit is passing

Reply:
oops I was double deleting the container. Fixed